docs/iam-usage.rst

```diff
@@ -0,0 +1,148 @@
+Using IAM For Users
+===================
+
+.. note:: This document assumes basic knowledge of Google Cloud IAM,
+   see `the docs <https://cloud.google.com/iam/docs/>`_ for details
+
+Resources that implement the IAM interface provide the following methods:
+
+Policy Editing Convenience Methods
+----------------------------------
+
+Common keyword arguments:
+
+- ``retries`` number of times to retry in the
+   even of an interrupted transaction, (not available in ``set_policy``)
+- ``version`` version number of the new policy. Not used for transactionality.
+   provided for end-user use. Left blank, will increment the version by 1.
+
+
+>>> resource.add_role(iam.user('alice@example.com'), iam.roles.OWNER.name)
+True
+
+The first argument is one of the IAM "Member" types:
+
+- ``iam.user(email)`` an individual Google account
+- ``iam.service_account(email)`` a Google Cloud Service Account
+- ``iam.group(email)`` a Google group.
+- ``iam.domain(domain_name)`` a Google apps domain
+- ``iam.ALL_AUTHENTICATED_USERS`` any authenticated user
+- ``iam.ALL_USERS``
+
+.. note:: that all of these are convenience wrappers around strings.
+   See the list of member string formats `here <https://cloud.google.com/iam/docs/managing-policies>`_.
+
+The second argument is a role to added to the specified member. Roles are simply strings that correspond to sets of permissions.
```

Comment:
This seems so far removed from "the first argument is", I forgot we were talking about arguments. Suggest discussing user types elsewhere and referring to it. Ditto with this part on separating the discussion of roles from the discussion of the argument.